gradle-conjure/src/main/java/com/palantir/gradle/conjure/GenerateNpmRcTask.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.conjure;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.google.common.net.HttpHeaders;
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodySubscribers;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Optional;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Parameter;
+
+public class GenerateNpmRcTask extends DefaultTask {
+    private static final JsonMapper MAPPER = ObjectMappers.newClientJsonMapper();
+
+    private static final String NPM_REGISTRY_URI_PROPERTY = "npmRegistryUri";
+    private final RegularFileProperty outputFile = getProject().getObjects().fileProperty();
+    private final Property<String> username = getProject().getObjects().property(String.class);
+    private final Property<String> password = getProject().getObjects().property(String.class);
+    private final Property<String> packageName = getProject().getObjects().property(String.class);
+
+    @OutputFile
+    public final RegularFileProperty getOutputFile() {
+        return this.outputFile;
+    }
+
+    @Input
+    public final Property<String> getUsername() {
```

Comment:
I haven't added explicit extension to configure them but happy to, this is not necessary for other languages so wasn't sure what structure we want them to follow